index.js

```diff
@@ -25,78 +25,48 @@ function fail (err) {
   return err;
 }
 
-function openConfig (cb) {
-  var userHome = require('user-home');
-  var configpath = path.join(userHome || os.tmpdir(), configfile);
-  // open file for reading and appending. if the filesize is zero
-  // we will spawn node with --v8-options and write the parsed
-  // options to a file. if it is larger than zero, we'll just read
-  // the file and be done.
-  fs.open(configpath, 'a+', function (err, fd) {
-    if (err) {
-      return cb(fail(err));
-    }
-    return cb(null, fd);
-  });
-}
-
-function writeConfig (fd, cb) {
+function getFlags(cb) {
   execFile(process.execPath, ['--v8-options'], function (execErr, result) {
     var flags;
     if (execErr) {
       return cb(execErr);
     }
+
     flags = result.match(/\s\s--(\w+)/gm).map(function (match) {
       return match.substring(2);
     }).filter(function (name) {
       return exclusions.indexOf(name) === -1;
     });
-    var buf = new Buffer(JSON.stringify(flags));
-    fs.write(fd, buf, 0, buf.length, 0, function (writeErr, bytesWritten, buffer) {
-      // linux ignores positional arguments when files are open in append mode.
-      // the truncate call below ensures that multiple concurrent processes
-      // trying to write to this config will not result in a file with the
-      // contents appended multiple times.
-      fs.ftruncate(fd, buf.length, function (truncErr) {
-        fs.close(fd, function (closeErr) {
-          var err = truncErr || writeErr || closeErr;
-          if (err) {
-            return cb(fail(err));
-          }
-          return cb(null, JSON.parse(buffer.toString()));
-        });
-      })
-    });
-  });
-}
 
-function readConfig (fd, filesize, cb) {
-  var buf = new Buffer(filesize);
-  fs.read(fd, buf, 0, filesize, 0, function (readErr, bytesRead, buffer) {
-    fs.close(fd, function (closeErr) {
-      var err = readErr || closeErr;
-      if (err) {
-        return cb(fail(err));
-      }
-      return cb(null, JSON.parse(buffer.toString()));
-    });
+    cb(flags);
   });
 }
 
+
 module.exports = function (cb) {
-  openConfig(function (err, fd) {
-    if (err) {
-      return cb(fail(err));
+  var userHome = require('user-home');
+  var configpath = path.join(userHome || os.tmpdir(), configfile);
+  fs.exists(configpath, function(e) {
```

Comment:
fs.exists is an antipattern, the file could come into existence between this call and the call to create the file.